test/model/internal_link_test.dart

```diff
@@ -9,6 +9,13 @@ import 'package:zulip/model/store.dart';
 import '../example_data.dart' as eg;
 import 'test_store.dart';
 
+// Using Set instead of List in to avoid any duplicated test urls.
+Set<String> getUrlSyntaxVariants(String urlString) {
+  final urlWithChannelSyntax = urlString.replaceFirst('#narrow/stream', '#narrow/channel');
+  final urlWithStreamSyntax = urlString.replaceFirst('#narrow/channel', '#narrow/stream');
```

Comment:
Let's relax this so it handles the `stream`/`channel` operator even if it's at some position other than the first. It's usually the first, but it doesn't have to be. For example, this is a valid narrow link that the web app knows how to handle: https://chat.zulip.org/#narrow/is/unread/stream/14-GSoC

I think this could be done like this:
- in `urlString`, skip past the `#narrow/` at the beginning
- split the rest by '/'
- in that list of segments, look at the even-indexed ones (i.e., `index % 2 == 0`)
- for each of those that's equal to "stream", replace it with "channel" and vice versa

(We wouldn't want to just replace all occurrences of "stream" or "channel" in the URL; we want to avoid doing things like destroying a topic operand that happens to have one of those strings in it.)

Reply:
It's true those are valid links, but it looks like we don't currently have any examples of them in this test file.

(Likely we should! Our implementation looks like it accepts them. But we don't right now. These tests were basically ported from zulip-mobile, because the tests there for this area were quite thorough, but that implementation doesn't accept those links.)

So for this PR, I'm happy to keep it simple in this respect.

Adding that logic would be a good followup, along with adding some test cases where /channel/ comes after /topic/, and where things come in other surprising orders.

Reply:
Cool, thanks for that suggestion, Greg. Since this was my only comment, I'll mark this for integration review. 🙂